### PR TITLE
[DURACOM-148] add crossref metadata processor to manage the date issued field

### LINKED ISSUE #8861: The CrossRef integration of Live Import Framework manage only the year for the date issued
**Describe the bug**
When trying to import an item from CrossRef, I noticed that the dc.date.issued only contains the year, even though the item also contains other data such as month or day.

**To reproduce**
1. Go to MyDSpace
2. Start the Import metadata from an external source
3. Enter the DOI `10.1629/22161` and specify CrossRef
4. Start the submission with the item
5. The Date of Issue field contains only the year, while the crossref entry https://api.crossref.org/works/10.1629/22161 also contains the month and day: 2009-01-01

A video demonstrating the problem follows:
[screencast-localhost_4000-2023.05.22-16_21_08.webm](https://github.com/DSpace/DSpace/assets/36161521/b8f131d6-aee1-43bf-b44b-193e4827f3dc)

**Expected behavior**
The CrossRef integration should also use the month and day to populate the Issue Date field.

### PR DESCRIPTION
**Pull request by francescopioscognamiglio**
## References
Fixes #8861

## Description
I added a new metadata processor to correctly populate the dc.date.issued metadata.

## Instructions for Reviewers

Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

List of changes in this PR:
* I added a new CrossRef metadata processor `CrossRefDateMetadataProcessor` that uses year, month and day to populate the dc.date.issued metadata
* I updated the tests in `CrossRefImportMetadataSourceServiceIT` to check the full issued date

**Include guidance for how to test or review your PR.**
1. Go to MyDSpace
2. Start the Import metadata from an external source
3. Enter the DOI `10.1629/22161` and specify CrossRef
4. Start the submission with the item
5. The Date of Issue field contains the full date `2009-01-01`

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [ ] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by alanorth** [comment]
I tested this and it works. I used `10.1186/s40781-015-0073-1`, which has a year, month, and date, and it was processed and populated in the input form successfully.

Crossref's date format is really annoying and requires custom processing! Thanks @francescopioscognamiglio. This is also a good example of designing a simple custom processor.

File: dspace-api/src/main/java/org/dspace/importer/external/crossref/CrossRefDateMetadataProcessor.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.importer.external.metadatamapping.contributor.JsonPathMetadataProcessor;
+import org.joda.time.LocalDate;
+
+/**
+ * This class is used for CrossRef's Live-Import to extract
+ * issued attribute.
+ * Beans are configured in the crossref-integration.xml file.
+ *
```
Comment (alanorth) [inline dspace-api/src/main/java/org/dspace/importer/external/crossref/CrossRefDateMetadataProcessor.java:27]: FYI this comment introduces trailing whitespace.

Comment (francescopioscognamiglio) [reply dspace-api/src/main/java/org/dspace/importer/external/crossref/CrossRefDateMetadataProcessor.java]: Hi @alanorth, I removed the trailing whitespace from the javadoc. Let me know, thank you!

Comment (alanorth) [reply dspace-api/src/main/java/org/dspace/importer/external/crossref/CrossRefDateMetadataProcessor.java]: Thanks @francescopioscognamiglio. I don't know if anyone has any other comments about the code, but it works as expected.

**Review by alanorth — APPROVED** [review]

**Pull request merged**